### ISSUE TITLE
number format

### ISSUE THREAD

**Issue by howi42** [kalkih/mini-graph-card#1010]
the number format, as defined in the profile settings, is not honored.

**Comment by ildar170975** [comment]
Same as closed https://github.com/kalkih/mini-graph-card/issues/774

**Comment by howi42** [comment]
Regardless the state of this issue, the user interface does not look good with this.
And , I have not found any settings combination ( language, number format, data format ), which  at the end use the same style for numbers.

**Comment by cataseven** [comment]
Up up up

**Comment by Lavve** [comment]
Would be really nice to get this fixed, it looks a bit strange with different number formats on the dash.

**Comment by CMDR-Sloma** [comment]
This should be treated as a bug, not an enhancement.

**Comment by cataseven** [comment]
Up

**Comment by ildar170975** [comment]
No need to "up" here. It only makes sense to prevent "stale".

**Comment by cataseven** [comment]
Can you please label this  as "bug". This is not an ench.

**Comment by CMDR-Sloma** [comment]
I also agree this is not an enhancement but a bug.

**Comment by ildar170975** [comment]
I can set here "bug", but it may not "speed up" implementation...

**Comment by cataseven** [comment]
> I can set here "bug", but it may not "speed up" implementation... So, 2 labels - enhancement & bug - everyone is happy))))

Maybe My last suggestion speed it up ;)

**Comment by aigarius** [comment]
Not really, cat. The addon should be following the Home Assistant defined settings on the Number Format *specifically* and not just use the default number format of the language chosen for the HA UI. And also not one random language format. For example, I have HA UI language set to "English", but the Number format to "None" because I can not stand having extra thousand separator symbols. It works (almost) everywhere in HA, but this addon just ignores the *very explicit* Number format setting and instead tries to guess the number format from the UI language, which is a bug.

**Comment by cataseven** [comment]
> Not really, cat. The addon should be following the Home Assistant defined settings on the Number Format _specifically_ and not just use the default number format of the language chosen for the HA UI. And also not one random language format. For example, I have HA UI language set to "English", but the Number format to "None" because I can not stand having extra thousand separator symbols. It works (almost) everywhere in HA, but this addon just ignores the _very explicit_ Number format setting and instead tries to guess the number format from the UI language, which is a bug.

Sure, but I at least some of us get what they need. Agree for the rest

**Comment by aigarius** [comment]
I wonder if https://github.com/home-assistant/frontend/pull/17249 can be used to get HA to format the values correctly.

**Comment by ildar170975** [comment]
May be I am missing something, see no need to connect “number format” and “language”, imho these are independent settings.

**Comment by ildar170975** [comment]
And in HA itself there are some “connections” between a language and a number format - which makes a code more complex and buggy.

**Comment by aigarius** [comment]
There are indeed, like if I click on a value I get this wonderful screen:
![Screenshot from 2024-02-20 15-00-46](https://github.com/kalkih/mini-graph-card/assets/295480/98aef427-20f5-4475-8685-6444c3609dc7)

That screen has **both** correct number formatting preference (None) in the number display location (top right, "1078 ppm"), but at the same time chooses to use the language numbering preference for the Y axis ("2,000")

The second is a bug.

**Comment by ildar170975** [comment]
What happens if you start playing with a language, will it affect mini-graph?

**Comment by aigarius** [comment]
Yes, switching from "English" to German, for example, would change the decimal separator from "." to "," and the thousand separator from "," to ".". And switching language to Esperanto would change the thousand separator to " " and decimal separator to ",".

The code in https://github.com/kalkih/mini-graph-card/blob/master/src/main.js#L720 literally just gives the value to the browser function Intl.NumberFormat along with the currently configured system language.

**Comment by aigarius** [comment]
https://developers.home-assistant.io/docs/frontend/data/#hassformatentityattributevaluestateobj-attribute-value describes the function that is there to format the value correctly, according to settings of both HA and of the entity itself (like you can set precision on the entity). It is available to mini-graph-card.

**Comment by dr3amr** [comment]
Any development here ? No updates from Feb 20, is this still considered for implementation ?

**Comment by Lavve** [comment]
Still no activity regarding this issue, or are there any updates coming anytime soon to fix this?

**Comment by ildar170975** [comment]
@dr3amr and others
> Any development to current issue ? No updates from Feb 20, is this still considered for implementation ?

Please stop urging developers. You will be informed about new features & bugfixes when a new Release arrives. It will happen when it happens. No promises unfortunately.

**Comment by ildar170975** [comment]
check v0.14

**Comment by Ascathon** [comment]
This seems to work fine for most of my Aqara temp sensors. Still figuring out why it doesn't work for my both Tado thermostat sensors. Precision via entity settings for these work fine for built in sensor cards.

**Comment by aigarius** [comment]
Yes, works now on v0.14-dev1
